src/OnboardingSPA/pages/Steps/GetStarted/SiteTypeSetup/SecondarySite/index.js

```diff
@@ -101,71 +102,75 @@ const StepPrimarySetup = () => {
+					<div className="nfd-setup-secondary-categories">
+						<div className="nfd-card-category-wrapper">
+							<div className="category-scrolling-wrapper">
+								<span
+									className="icon"
+									style={ {
+										backgroundImage: categoriesArray[ 0 ].icon,
+									} }
+								/>
+								<p className="categName">
+									{ ' ' }
+									{ categoriesArray[ 0 ].name }
+								</p>
+							</div>
 						</div>
-					</div>
-
-					<div className="subCategoriesSection">
-						{ categoriesArray[ 0 ]?.subCategories?.map(
-							( item, idx ) => {
-								return (
-									<span
-										key={ item }
-										onClick={ ( e ) =>
-											handleCategoryClick( idx )
-										}
-										className={ `${
-											clickedIndex === idx ||
-											item === selectedCategoryInStore
-												? 'chosenSecondaryCategory '
-												: ''
-										}nfd-card-category` }
-									>
-										{ item }
-									</span>
-								);
-							}
-						) }
-					</div>
-				</div>
 
-				<div className="nfd-setup-primary-second">
-					<div className="nfd-setup-primary-second-top">
-						<div className="blackText">
-							{ __(
-								content.tellusHereText,
-								'wp-module-onboarding'
+						<div className="subCategoriesSection">
+							{ categoriesArray[ 0 ]?.subCategories?.map(
+								( item, idx ) => {
+									return (
+										<span
+											key={ item }
+											onClick={ ( e ) =>
+												handleCategoryClick( idx )
+											}
+											className={ `${
+												clickedIndex === idx ||
+												item === selectedCategoryInStore
+													? 'chosenSecondaryCategory '
+													: ''
+											}nfd-card-category` }
+										>
+											{ item }
+										</span>
+									);
+								}
 							) }
 						</div>
-						<input
-							type="text"
-							onChange={ ( e ) => categoryInput( e ) }
-							className="tellUsInput"
-							placeholder={ sprintf(
-								__(
-									content.placeholderSiteTypeInput,
+					</div>
+
+					<div className="nfd-setup-primary-second">
+						<div className="nfd-setup-primary-second-top">
+							<div className="blackText">
+								{ __(
+									content.tellusHereText,
 									'wp-module-onboarding'
-								),
-								translations( 'site' )
-							) }
-							value={ inputCategVal }
-						/>
+								) }
+							</div>
+							<input
+								type="text"
+								onChange={ ( e ) => categoryInput( e ) }
+								className="tellUsInput"
+								placeholder={ sprintf(
+									__(
+										content.placeholderSiteTypeInput,
+										'wp-module-onboarding'
+									),
+									translations( 'site' )
+								) }
+								value={ inputCategVal }
+							/>
+						</div>
 					</div>
-				</div>
-				<NavCardButton text={ __( content.buttonText ) } />
+				</StepStateHandler>
+				<NavCardButton
+					text={ __( content.buttonText ) }
+					disabled={ categoriesArray[ 0 ]?.subCategories === null }
+				/>
```

Comment:
We could include these tags within the StepStateHandler tag too

Reply:
we can .. but then the button will also not appear when the page loads .. it will appear after a delay of 500ms ... we will have to take a call on that whether we want to do that with the buttons also

Reply:
We can ask David but I don't think it's a good idea to move the `NavCardButton` and the `NeedHelpTag` into the handler because these are not dependent on the data that is being waited for and hence do not need the delay, if navigation is the concern then let me remind you that the user can always navigate using the header navigation regardless. We also do not want to hide the need help tag incase something breaks and the component does not fade in its children.